src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPEditorFeatureDetector.cs

```diff
@@ -16,11 +17,16 @@ internal class DefaultLSPEditorFeatureDetector : LSPEditorFeatureDetector
     {
         private const string DotNetCoreCSharpCapability = "CSharp&CPS";
         private const string RazorLSPEditorFeatureFlag = "Razor.LSP.Editor";
+
         private static readonly Guid LiveShareHostUIContextGuid = Guid.Parse("62de1aa5-70b0-4934-9324-680896466fe1");
         private static readonly Guid LiveShareGuestUIContextGuid = Guid.Parse("fd93f3eb-60da-49cd-af15-acda729e357e");
+
         private readonly ProjectHierarchyInspector _projectHierarchyInspector;
         private readonly Lazy<IVsUIShellOpenDocument> _vsUIShellOpenDocument;
         private readonly IVsFeatureFlags _featureFlags;
+
+        private readonly Dictionary<string, bool> _projectSupportsRazorLSPCache;
```

Comment:
Hmm, so you'll probably want to utilize a memory cache for this because as it's currently implemented the cache will never be cleaned up and will grow in memory as a user navigates through projects.

A memory cache would have a set size and upon reaching n number of cache items would start evicting old ones. For instance, @ryanbrandenburg added a copy of the memory cache available by the runtime here: https://github.com/dotnet/aspnetcore-tooling/blob/06625ac6b98599f7fe1902bbe1ef59ce165469ba/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/MemoryCache.cs#L13

In your case you might be able to use the true memory cache implementation.

Reply:
Fair enough. Assuming max path length of 260 chars + 1 char for the bool, and 100 files, we'd be taking up ~25MB (assuming no overhead for the `Dictionary`. I'll go ahead with the memcache.

Reply:
@TanayParikh Yeah I believe since you're in a Microsoft.VisualStudio project you have access to the "real" cache which would be far preferable to my hacked together one.

Reply:
Latest commit should in theory work, however it's crashing the entire LSP experience on boot. 

```
LimitedFunctionality
System.AggregateException: Project system data flow 'UnconfiguredProjectEvaluationGlobalPropertiesAggregator: 6307725' closed because of an exception.
An exception was thrown while initializing part "Microsoft.VisualStudio.LanguageServerClient.Razor.DefaultLSPEditorFeatureDetector". ---> Microsoft.VisualStudio.Composition.CompositionFailedException: An exception was thrown while initializing part "Microsoft.VisualStudio.LanguageServerClient.Razor.DefaultLSPEditorFeatureDetector". ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.Caching.Memory, Version=3.1.5.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified.
   at Microsoft.VisualStudio.LanguageServerClient.Razor.DefaultLSPEditorFeatureDetector..ctor(ProjectHierarchyInspector projectHierarchyInspector)
```

Have you seen this before?

Reply:
Need to add Microsoft.Extensions.Caching.Memory to the VS extensions codebase list: https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.VisualStudio.RazorExtension/AssemblyCodeBases.cs

Reply:
Thanks [that](https://github.com/dotnet/aspnetcore-tooling/pull/2235/commits/4852d50b91a461cbdb6c799e89ffb446a497b7b0) did it! Do we need to update component governance with this - or is that for non-Microsoft components?

Reply:
Will need to update. That being said it's easy

Reply:
https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.VisualStudio.RazorExtension/NOTICE.txt#L21

https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.VisualStudio.RazorExtension/NOTICE.txt#L127

Looks like we already have it? Do the versions need to be updated?

Reply:
@NTaylorMullen just confirming the above (existing) references are sufficient?